hashintel/hash
# Setup MUI

## 🌟 What is the purpose of this PR?


Sets up MUI and creates a theme based on our design system

## 🔍 What does this change?

- Sets up MUI in the frontend repo
- Adds colors, typography, shadows, border radii based on the design system.
- Adds FontAwesomeSvgIcon component for rendering font-awesome icons


## 🔗 Related links





- [Introduce MUI + theme to hash repo](https://app.asana.com/0/1201825402400452/1201858717936365) (_internal_)

## 🛡 What tests cover this?



- Manual

## ❓ How to test this?



1.  Install the latest dependencies and start the app
2.  Add the following to any page in the app (preferably login page if you aren't authenticated)
```
import { Box } from "@mui/material"

<Box sx={{
       height: 400,
       width: 400,
       backgroundColor: ({ palette }) => palette.purple[60],
       borderRadius: ({ borderRadii }) => borderRadii.md,
       boxShadow: ({ shadows }) => shadows.lg
  }} />
<Typography variant="title">Title</Typography>
```
3.  Confirm that it renders

## 📹 Demo
